_posts/2024-09-04-open-energy-models.md

```diff
@@ -0,0 +1,606 @@
+system operational problems at hourly resolution via a rolling horizon model of
+48 hour blocks rolling forward 24 hours at each step.
+
+Diego's experience reinforces for me the need to include rolling horizon
+simulations in our set of benchmark instances that we are building in the
+[jump-dev/open-energy-modeling-benchmarks](https://github.com/jump-dev/open-energy-modeling-benchmarks)
+repository.
+
+[_Back to contents_](#contents)
+
+## [2024] Solving the Market-to-Market Problem in Large Scale Power Systems
+
+_Speaker: Jose Daniel Lara @jd-lara_
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/N-jDHickaTc?si=T7eQnTopbCCtES3X" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
+
+In this talk Jose gave an overview and spoke about recent updates to
+[Sienna](https://www.nrel.gov/analysis/sienna.html) (nee PowerSimulations.jl).
+The talk follows his previous talks at [JuMP-dev 2019](#2019-powersimulationsjl)
+and [JuMP-dev 2023](https://www.youtube.com/watch?v=J7VbCKsnTvQ).
+
+A key feature of Sienna is that it is designed for problems with _O(10⁸)_
+variables and constraints. They are building and solving simulation models of
+the Eastern Interconnection, which is one of, it not the, largest power system
+in the world (it has 150,000 buses and 270,000 lines).
+
+A key lesson that I took from Jose's talk is that they have already encountered
+(and resolved or worked around) many of same issues that Diego raised in his
+talk, such as how to efficiently save the large volume of data that comes out of
+the simulation models.
+
+In regard to feature requests for future versions of JuMP, Jose again referenced
+the need for efficient re-solves of optimization models with parameters. We
+have, over the years, experimented with a number of different approaches in
+Sienna, including the now largely defunct [ParameterJuMP.jl](https://github.com/JuliaStochOpt/ParameterJuMP.jl),
+the intended replacement [ParametricOptInterface.jl](https://github.com/jump-dev/ParametricOptInterface.jl),
+and other work-arounds like adding new `VariableIndex in EqualTo` constraints.
+Ultimately, each way has a different trade-off (and it also depends on the size
+of the model that you are trying to solve), and we haven't found something that
+we are completely happy with.
+
+My main takeaways are that we should include benchmarks that use
+ParametricOptInterface.jl in our [jump-dev/open-energy-modeling-benchmarks](https://github.com/jump-dev/open-energy-modeling-benchmarks)
+repository, and that JuMP's ability to efficiently solve a sequence of similar
+problems with some changes in the problem parameters is both a main selling
+point of using JuMP over alternative tools and a current bottleneck in existing
+workflows.
```

Comment:
Sienna is also one of the few tools that has a workflow with multiple resolves as part of the features and needs since it is the only JuMP application focused on simulations.

Reply:
I don't know if this is true. SpineOpt has a rolling-horizon feature, and so does @sstromer's model.

Reply:
The same is currently in development for EnergyModelsX.